cmd/node-cache/metrics.go

```diff
@@ -0,0 +1,46 @@
+package main
+
+import (
+	"net"
+	"net/http"
+
+	"github.com/coredns/coredns/plugin"
+	clog "github.com/coredns/coredns/plugin/pkg/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+var (
+	setupErrCount = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Namespace: plugin.Namespace,
+		Subsystem: "nodecache",
+		Name:      "setup_errors",
+		Help:      "The number of errors during periodic network setup for node-cache",
+	}, []string{"errortype"})
+)
+
+func initMetrics(ipport string) {
+	serveMetrics(ipport)
```

Comment:
we should not eat the error here?

Reply:
Fixed by checking error here and logging.